src/main/java/org/threadly/concurrent/CentralThreadlyPool.java

```diff
@@ -0,0 +1,215 @@
+    if (singleThreaded) {
+      return SINGLE_THREADED_LOW_PRIORITY_POOL;
+    } else {
+      return LOW_PRIORITY_POOL;
+    }
+  }
+  
+  /**
+   * Return a single threaded pool.  This can be useful for submitting tasks on where you don't 
+   * want to worry about any concurrency or shared memory issues.
+   * 
+   * @param threadGuaranteed {@code true} indicates that the pool manager needs to expand its size if necessary
+   * @return Single threaded pool for running or scheduling tasks on
+   */
+  public static PrioritySchedulerService singleThreadPool(boolean threadGuaranteed) {
+    return new NoThreadSchedulerRunningTask(TaskPriority.High, threadGuaranteed);
+  }
+  
+  /**
+   * Requests a pool with a given range of threads.  Minimum threads are guaranteed to be available 
+   * to tasks submitted to the returned threads.  Maximum threads help prevent tasks submitted to 
+   * this pool from possibly fully consuming the central pool.  If {@code maxThreads <= 0} then 
+   * tasks submitted to this pool will be able to use all available threads as long as they don't 
+   * end up consuming threads which other pools might need reserved.
+   * 
+   * @param minThreads Minimum number of threads the provided pool should be guaranteed to have, negative is ignored
+   * @param maxThreads Maximum number of threads to the returned pool can consume, or negative to use any available
+   * @return A pool with the requested specifications for task scheduling or execution
+   */
+  public static SchedulerService rangedThreadCountPool(int minThreads, int maxThreads) {
+    if (maxThreads == 1) {
+      return singleThreadPool(minThreads > 0);
+    } else {
+      return new GuaranteedConnectionProtectingScheduler(MASTER_SCHEDULER.getDefaultPriority(), 
+                                                         minThreads, maxThreads);
+    }
+  }
+  
+  private static PrioritySchedulerService masterScheduler(TaskPriority defaultPriority) {
+    if (defaultPriority == TaskPriority.High) {
+      return MASTER_SCHEDULER;
+    } else {  // starvable is assumed to not be used in this current implementation
+      return LOW_PRIORITY_MASTER_SCHEDULER;
+    }
+  }
+  
+  protected static class PoolResizer {
+    private final int amount;
+    
+    public PoolResizer(int amount) {
+      this.amount = amount;
+
+      MASTER_SCHEDULER.adjustPoolSize(amount);
+      GUARANTEED_THREADS.add(amount);
+    }
+    
+    @Override
+    protected void finalize() {
+      GUARANTEED_THREADS.add(-amount);
+      MASTER_SCHEDULER.adjustPoolSize(-amount);
+    }
+  }
+
+  // TODO - this is slightly flawed, we need to have a way for the pool to grow without 
+  //          requiring protected threads otherwise we will never use available resources.
+  //          If it does not make sense changing when the pool is expanded (or how), then this 
+  //          class is probably providing a protection that makes everything useless...quite a thinker
```

Comment:
Yeah, The only other way I can think to watch this is for threads to have some kind of processing timer we check, or have a recurring task we watch for getting blocked, both have different issues, mainly they could grow like crazy if load is high even in situations where threads are not stuck in blocking operations they would just grow different in different situations.